packages/workflow/src/records/handler/archive.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * OpenCRVS is also distributed under the terms of the Civil Registration
+ * & Healthcare Disclaimer located at http://opencrvs.org/license.
+ *
+ * Copyright (C) The OpenCRVS Authors located at https://github.com/opencrvs/opencrvs-core/blob/master/AUTHORS.
+ */
+import * as z from 'zod'
+import * as Hapi from '@hapi/hapi'
+import { getLoggedInPractitionerResource } from '@workflow/features/user/utils'
+import { getToken } from '@workflow/utils/authUtils'
+import { validateRequest } from '@workflow/utils/index'
+import { toArchived } from '@workflow/records/state-transitions'
+import { sendBundleToHearth } from '@workflow/records/fhir'
+import { indexBundle } from '@workflow/records/search'
+import { getRecordById } from '@workflow/records/index'
+
+const requestSchema = z.object({
+  id: z.string(),
+  reason: z.string().optional(),
+  comment: z.string().optional(),
+  duplicateTrackingId: z.string().optional()
+})
+
+export async function archiveRecordHandler(
+  request: Hapi.Request,
+  _: Hapi.ResponseToolkit
+) {
```

Comment:
Can we use the "createRoute" function here? We could add a optional parameter about including history records when downloading the record within it